pkg/controllers/user/helm/common/extra_args_test.go

```diff
@@ -0,0 +1,35 @@
+package common
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/types/apis/project.cattle.io/v3"
+)
+
+func Test_injectDefaultRegistry(t *testing.T) {
+	testRegistry := "test.registry.com"
+	if err := settings.SystemDefaultRegistry.Set(testRegistry); err != nil {
+		t.Fatalf("failed to set system default registry settings, %s", err.Error())
+	}
+
+	testCases := map[string]bool{
+		settings.SystemMonitoringCatalogID.Get():                       true,
+		"catalog://?catalog=library&template=wordpress&version=2.1.11": false,
+	}
+
+	for externalID, shouldMatch := range testCases {
+		testApp := &v3.App{
+			Spec: v3.AppSpec{
+				ExternalID: externalID,
+			},
+		}
+		v, ok := injectDefaultRegistry(testApp)["systemDefaultRegistry"]
```

Comment:
Don't access the map directly off the function call, this makes it unclean what is actually happening here